movie-service/src/main/java/com/garret/movies/service/impl/MovieServiceImpl.java

```diff
@@ -0,0 +1,135 @@
+package com.garret.movies.service.impl;
+
+import com.garret.movies.dao.entity.Movie;
+import com.garret.movies.dao.repository.MovieRepository;
+import com.garret.movies.service.api.MovieService;
+import com.garret.movies.service.exception.IncorrectDateException;
+import com.google.inject.internal.util.ImmutableList;
+import lombok.AllArgsConstructor;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Slf4j
+@AllArgsConstructor
+public class MovieServiceImpl implements MovieService {
+
+    private MovieRepository movieRepository;
+
+    @Override
+    @Transactional
+    public Movie save(@NonNull Movie movie) {
+        Optional<Movie> optionalMovie = movieRepository.findByImdbId(movie.getImdbId());
+        if (optionalMovie.isPresent()) {
+            log.info(movie.getTitle() + " already present");
+            return optionalMovie.get();
+        }
+        log.info("Start saving movie " + movie.getTitle());
+        Movie savedMovie = movieRepository.save(movie);
+        log.info(movie.getTitle() + " was saved to database");
+        return savedMovie;
+    }
+
+    @Override
+    @Transactional
+    public List<Movie> saveAll(@NonNull List<Movie> movies) {
+        movieRepository.saveAll(movies);
+        log.info(movies.size() + " movies saved to database");
+        return movies;
```

Comment:
@Garret-vin Its better to return result of save method as result movies will contain ids which was generated during persist. Returning input data not super useful as you already have it in place where you call this service